SessionPlayer.js

```diff
@@ -0,0 +1,198 @@
+                        },
+                        {
+                            type: 'Normal'
+                            stageDuration: [4, 2, 4, 0]
+                        },
+                        {
+                            type: 'Normal'
+                            stageDuration: [4, 2, 4, 0]
+                        },
+                    ]
+                },
+
+                { // Guides
+                    guideName: 'PowerPose',
+                    instructions: [ // Instructions
+                        { // Descriptor
+                            type: 'SuperHuman',
+                            stageDuration: [15],
+                        },
+                        {
+                            type: 'Star',
+                            stageDuration: [20],
+                        }
+                    ]
+                }
+            ];
+
+        this.state = {
+            guideState: 'playing', // 'playing', 'paused', 'finished'
+            currentGuideIndex: 0,
+            currentInstructionIndex: 0,
+            currentGuide: this.session[0].guide,
+            currentInstruction: this.sessionp[0].instructions[0],
+        };
+    }
+
+    onInstructionEndHandler = () => {
+        incrementInstructionOrGuide();
+    }
+
+    incrementInstructionOrGuide = () => {
+        const nextInstruction =
+            this.session[currentGuideIndex]
+                .instructions[currentInstructionIndex + 1];
+
+        const nextGuide = this.session[currentGuideIndex + 1];
+
+        const {
+            currentInstructionIndex,
+            currentGuideIndex,
+            currentGuide,
+            currentInstruction
+        } = this.state;
+
+        if (typeof nextInstruction !== 'undefined') {
+            const nextInstructionIndex = currentInstructionIndex + 1;
+
+            this.setState({
+                currentInstructionIndex: nextInstructionIndex,
+                currentInstruction: currentGuide.instructions[nextInstructionIndex],
+            });
+        } else if (typeof nextGuide !== 'undefined') {
+            const nextGuideIndex = currentGuideIndex + 1;
+
+            this.setState({
+                currentInstructionIndex: 0,
+                currentInstruction: nextGuide.instructions[0],
+                currentGuideIndex: nextGuideIndex,
+                currentGuide: nextGuide,
+            });
+        } else {
+            console.log('All guides have been completed');
+        }
+    }
+
+    actionButtonHandler = () => {
+        const { guideState } = this.state;
+
+        switch(guideState) {
+            case 'playing':
+                this.setState({ guideState: 'paused' });
+                break;
+            case 'paused':
+                this.setState({ guideState: 'playing' });
+                break;
+            case 'finished':
+                incrementInstructionOrGuide();
+                break;
+        }
+    }
+
+    render() {
+        const {
```

Comment:
From `135` to `152`, you completely lost me. I don't like the `currentGuide` name repetition and the fact that you are trying to declare `const` 3 times.